mscore/scoreitemmodel.cpp

```diff
@@ -0,0 +1,136 @@
+//=============================================================================
+//  ScoreItemModel
+//
+//  Copyright (C) 2020 Peter Jonas
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#include "scoreitemmodel.h"
+
+#include "libmscore/score.h"
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   ScoreItemModel::ScoreItemModel
+//---------------------------------------------------------
+
+ScoreItemModel::ScoreItemModel(Score* score, QObject* parent)
+    : QAbstractItemModel(parent)
+{
+    _scoreRoot = score;
+    Q_ASSERT(_scoreRoot);
+}
+
+//---------------------------------------------------------
+//   ScoreItemModel::elementFromIndex
+//---------------------------------------------------------
+
+ScoreElement* ScoreItemModel::scoreElementFromIndex(const QModelIndex& index) const
+{
+    if (index.isValid()) {
+        return static_cast<ScoreElement*>(index.internalPointer());
+    } else {
+        return _scoreRoot;
+    }
+}
+
+//---------------------------------------------------------
+//   ScoreItemModel::data
+//---------------------------------------------------------
+
+QVariant ScoreItemModel::data(const QModelIndex& index, int role) const
+{
+    Q_ASSERT(index.column() == 0 || !index.isValid());
+    ScoreElement* scoreElement = scoreElementFromIndex(index);
+    switch (role) {
+    case Qt::DisplayRole:
+    case Qt::ToolTipRole:
+    case Qt::StatusTipRole:
+    case Qt::WhatsThisRole:
+        return scoreElement->userName();
+    case Qt::DecorationRole:
+    case Qt::EditRole:
+    case Qt::SizeHintRole:
+    default:
+        return QVariant();
+    }
+}
+
+//---------------------------------------------------------
+//   ScoreItemModel::index
+// Return index to child at given row & column of parent index.
```

Comment:
Please add an extra slash `/` to this line (i.e. `///`) to expose this comment to Doxygen. It should be:

```c++
//---------------------------------------------------------
//   ScoreItemModel::index
/// Return index to child at given row & column of parent index.
//---------------------------------------------------------
```

Please do the same for the other doc-style comments in the file.